application/src/main/java/org/togetherjava/tjbot/features/moderation/TransferQuestionCommand.java

```diff
@@ -0,0 +1,182 @@
+                    .build();
+
+        event.replyModal(transferModal)
+            .queue(success -> logger.info(
+                    "{} with id: {}  triggered the transfer action on message with id: {}",
+                    event.getUser().getName(), event.getUser().getId(), originalMessageId),
+                    failed -> {
+                    });
+    }
+
+    @Override
+    public void onModalSubmitted(ModalInteractionEvent event, List<String> args) {
+        event.getJDA()
+            .retrieveUserById(args.get(0))
+            .map(user -> user)
+            .flatMap(fetchedUser -> createForumPost(event, fetchedUser))
+            .flatMap(user -> dmUser(event.getChannel(), user))
+            .flatMap(helperForum -> event.reply("transferred to" + helperForum.getAsMention())
+                .setEphemeral(true))
+            .flatMap(info -> deleteOriginalMessage(event.getJDA(), args.get(2), args.get(1)))
+            .queue();
+    }
+
+    private static String createTitle(String message) {
+        if (message.length() >= TITLE_MAX_LENGTH) {
+            int lastWordEnd = message.lastIndexOf(' ', TITLE_MAX_LENGTH);
+
+            if (lastWordEnd == -1) {
+                lastWordEnd = TITLE_MAX_LENGTH;
+            }
+
+            message = message.substring(0, lastWordEnd);
+        }
+
+        return isTitleValid(message) ? message : "Untitled";
+    }
+
+    private static boolean isTitleValid(CharSequence title) {
+        String titleCompact = TITLE_COMPACT_REMOVAL_PATTERN.matcher(title).replaceAll("");
+
+        return titleCompact.length() >= TITLE_COMPACT_LENGTH_MIN
+                && titleCompact.length() <= TITLE_COMPACT_LENGTH_MAX;
+    }
+
+    private RestAction<User> createForumPost(ModalInteractionEvent event, User originalUser) {
+        MessageCreateData forumMessage = MessageCreateData
+            .fromContent(event.getValue(TRANSFER_QUESTION_INPUT_ID).getAsString());
+        String forumTitle = event.getValue(TRANSFER_QUESTION_TITLE_ID).getAsString();
+        ForumChannel questionsForum = getHelperForum(event.getJDA());
+        List<ForumTag> tags = questionsForum.getAvailableTagsByName("JAVA", true);
+        ForumTag javaTag = tags.get(0);
+
+        return questionsForum.createForumPost(forumTitle, forumMessage)
+            .setTags(ForumTagSnowflake.fromId(javaTag.getId()))
+            .map(ForumPost::getMessage)
+            .flatMap(message -> message.reply("Original Post from " + originalUser.getAsMention()))
+            .map(sent -> originalUser);
+
+    }
+
+    private RestAction<ForumChannel> dmUser(MessageChannelUnion sourceChannel, User originalUser) {
+
+        return originalUser.openPrivateChannel()
+            .flatMap(channel -> channel.sendMessage(
+                    "Hi, TJ here. You are getting this message " + "because you tried to ask a "
+                            + "question in wrong chat, please read rules - Together Java"))
```

Comment:
y the random break in string? also u can't just hard code the server name

Reply:
My bad, i thought all the lines of code should split up to not go beyond a certain limit horizontally, but i think it's bad for logger ?

Reply:
u can just leave that on  spotless

Reply:
ohh now i see what ur doing, nvm mb

Reply:
is this better?
![image](https://github.com/Together-Java/TJ-Bot/assets/61616007/ede5e2dc-1c4e-4251-8e4c-59c4b9c467cd)